### ISSUE TITLE
Post-push CI build errors starting 2020-05-02

### ISSUE THREAD

**Issue by bartlettroscoe** [trilinos/Trilinos#7322]
As shown in [this query](https://testing-dev.sandia.gov/cdash/index.php?project=Trilinos&begin=2020-04-28&end=2020-05-04&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=61&value1=Linux-GCC-4.8.4-MPI_RELEASE_DEBUG_SHARED_PT_OPENMP_CI&field2=groupname&compare2=61&value2=Continuous) showing:

Site | Build Name | Revision | Time | Error | Warn | Time | Error | Warn | Time | Not Run | Fail | Pass | Time | Proc Time | Start Time | Labels 
 -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
ceerws1113 | Linux-GCC-4.8.4-MPI_-RELEASE_-DEBUG_-SHARED_-PT_-OPENMP_-CI | 84d69f | 6s | 0 | 0 | 3m 3s | 3 | 3 | 4m 39s | 3 | 1 | 3078 | 30m 38s | 4h 2m 50s | May 04, 2020 - 15:47 MDT | (54 labels)
ceerws1113 | Linux-GCC-4.8.4-MPI_-RELEASE_-DEBUG_-SHARED_-PT_-OPENMP_-CI | 48e13d | 6s | 0 | 0 | 3m 1s | 3 | 3 | 15m 6s | 3 | 0 | 3079 | 30m 38s | 4h 2m 47s | May 04, 2020 - 13:49 MDT | (54 labels)
ceerws1113 | Linux-GCC-4.8.4-MPI_-RELEASE_-DEBUG_-SHARED_-PT_-OPENMP_-CI | ac85ec | 6s | 0 | 0 | 3m 2s | 3 | 0 | 3m 36s | 3 | 0 | 3078 | 30m 44s | 4h 3m 32s | May 04, 2020 - 10:54 MDT | (54 labels)
ceerws1113 | Linux-GCC-4.8.4-MPI_-RELEASE_-DEBUG_-SHARED_-PT_-OPENMP_-CI | 04609c | 6s | 0 | 0 | 3m 1s | 3 | 0 | 5m 52s | 3 | 0 | 3078 | 30m 45s | 4h 3m 43s | May 04, 2020 - 09:39 MDT | (54 labels)
ceerws1113 | Linux-GCC-4.8.4-MPI_-RELEASE_-DEBUG_-SHARED_-PT_-OPENMP_-CI | b323f7 | 6s | 0 | 0 | 3m 15s | 3 | 0 | 4m 28s | 3 | 0 | 3078 | 30m 59s | 4h 5m 32s | May 04, 2020 - 07:48 MDT | (54 labels)
ceerws1113 | Linux-GCC-4.8.4-MPI_-RELEASE_-DEBUG_-SHARED_-PT_-OPENMP_-CI | 8c66f1 | 6s | 0 | 0 | 3m 51s | 3 | 50 | 57m 45s | 3 | 1 | 3077 | 30m 44s | 4h 3m 31s | May 02, 2020 - 00:14 MDT | (54 labels)
ceerws1113 | Linux-GCC-4.8.4-MPI_-RELEASE_-DEBUG_-SHARED_-PT_-OPENMP_-CI | cfd767 | 6s | 0 | 0 | 2m 38s | 0 | 0 | 0s |   |   |   |   |   | Apr 28, 2020 - 17:43 MDT | (29 labels)
ceerws1113 | Linux-GCC-4.8.4-MPI_-RELEASE_-DEBUG_-SHARED_-PT_-OPENMP_-CI | 2192fd | 6s | 0 | 0 | 3m 55s | 0 | 50 | 1h 59m 11s | 0 | 0 | 3079 | 30m 41s | 4h 3m 14s | Apr 28, 2020 - 03:53 MDT | (54 labels)

the post push CI build on 2020-05-02 showed 3 ROL build errors:

Subproject | Errors
-- | --
ROL | Error building packages/rol/test/sol/ROL_test_sol_checkSpectralRisk.exe
ROL | Error building packages/rol/test/sol/ROL_test_sol_solInterface.exe
ROL | Error building packages/rol/test/sol/ROL_test_sol_checkSuperQuantileQuadrangle.exe

showing errors like:

```
packages/rol/test/sol/CMakeFiles/ROL_test_sol_checkSpectralRisk.dir/test_09.cpp.o: file not recognized: File truncated
collect2: error: ld returned 1 exit status
```

From looking at:

* https://stackoverflow.com/questions/5713894/file-not-recognized-file-truncated-gcc-error

it says:

> Just remove the object file.
>
> This error most likely appeared after the previous build was interrupted and object file was not generated completely.

**Comment by bartlettroscoe** [comment]
To see if this could be cleared up from a build-from-scratch, I restarted the post-push CI server on 'ceerws1113' with:

```
$ cd /scratch/rabartl/Trilinos.base/SEMSCIBuild/

$  nohup env CTEST_BUILD_FLAGS="-j8 -k 999999" \
     CTEST_PARALLEL_LEVEL=8 \
    TRILINOS_CI_DO_INITIAL_REBUILD=1 \
  ./Trilinos/cmake/ctest/drivers/sems_ci/trilinos_ci_sever.sh \
    &> trilinos_ci_server.out &
```

and it looks like that it resolved the problem with the first full CI build as shown [here](https://testing-dev.sandia.gov/cdash/index.php?project=Trilinos&begin=2020-04-28&end=2020-05-06&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=61&value1=Linux-GCC-4.8.4-MPI_RELEASE_DEBUG_SHARED_PT_OPENMP_CI&field2=groupname&compare2=61&value2=Continuous) shown below:

Site | Build Name | Revision | Time | Error | Warn | Time | Error | Warn | Time | Not Run | Fail | Pass | Time | Proc Time | Start Time | Labels 
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
ceerws1113 | Linux-GCC-4.8.4-MPI_-RELEASE_-DEBUG_-SHARED_-PT_-OPENMP_-CI | b9b3fa | 6s | 0 | 0 | 47s | 0 | 0 | 26s | 0 | 0 | 169 | 6m 19s | 50m 27s | 1 hour ago | Panzer
ceerws1113 | Linux-GCC-4.8.4-MPI_-RELEASE_-DEBUG_-SHARED_-PT_-OPENMP_-CI | 48ffeb | 6s | 0 | 0 | 55s | 0 | 0 | 1m 16s | 0 | 0 | 175 | 6m 52s | 54m 54s | 4 hours ago | (2 labels)
ceerws1113 | Linux-GCC-4.8.4-MPI_-RELEASE_-DEBUG_-SHARED_-PT_-OPENMP_-CI | 7a7421 | 6s | 0 | 0 | 3m 10s | 0 | 50 | 2h 40m 37s | 0 | 0 | 3082 | 32m 6s | 4h 10m 32s | 18 hours ago | (54 labels)
ceerws1113 | Linux-GCC-4.8.4-MPI_-RELEASE_-DEBUG_-SHARED_-PT_-OPENMP_-CI | 84d69f | 6s | 0 | 0 | 3m 3s | 3 | 3 | 4m 39s | 3 | 1 | 3078 | 30m 38s | 4h 2m 50s | May 04, 2020 - 15:47 MDT | (54 labels)

**Comment by bartlettroscoe** [comment]
Looks like the post-push CI server is back in business.

It looks like I updated the cron driver on 'ceerws1113' on Saturday 2020-05-02 and it is possible that killed the CI build in progress by accident that may have resulted in partially written object files in ROL.

In any case, doing rebuilds with CMake may not be robust if one of the builds is killed in the middle of compiling into object files.  I will have to keep this in mind.

NOTE: The successful CI build iterations not sending out emails lead to:

* https://gitlab.kitware.com/snl/project-1/-/issues/142

Closing this issue as complete.
